### PR TITLE
Updates to azure module doc

### PR DESCRIPTION
**Pull request by karenzone**
Prerelease doc updates

### CODE REVIEW

File: docs/static/azure-module.asciidoc [hunk 1]
```diff
@@ -50,98 +50,53 @@ These {kib} dashboards are available and ready for you to use. You can use them
...
 
-You can specify <<azure_config_options,options>> for the Logstash Azure module in the
-`logstash.yml` configuration file or with overrides through the command line. 
-
-* *Command line configuration.* You can pass configuration options and run the module from the command line.
-The command line configuration uses the configuration options you provide and supports only one Event Hub.
+Specify <<azure_config_options,options>> for the Logstash Azure module in the
+`logstash.yml` configuration file. 
 
 * *Basic configuration.* You can use the `logstash.yml` file to configure inputs from multiple Event Hubs that share the same configuration.
```
Comment (acchen97) [inline docs/static/azure-module.asciidoc]: Can we make it clear here this is the option most users will use?

File: docs/static/azure-module.asciidoc [hunk 2]
```diff
@@ -50,128 +52,97 @@ These {kib} dashboards are available and ready for you to use. You can use them
...
-    var.kibana.host: "localhost:5601"
-    var.input.azure_event_hubs.threads: 8
-    var.input.azure_event_hubs.decorate_events: true
-    var.input.azure_event_hubs.consumer_group: "logstash"
-    var.input.azure_event_hubs.storage_connection: "DefaultEndpointsProtocol=https;AccountName=example...."
+    var.elasticsearch.hosts: localhost:9200
+    var.kibana.host: localhost:5601
+    var.input.azure_event_hubs.consumer_group: "logstash" <1>
+    var.input.azure_event_hubs.storage_connection: "DefaultEndpointsProtocol=https;AccountName=instance1..." <2>
+    var.input.azure_event_hubs.threads: 9
```
Comment (jakelandis) [inline docs/static/azure-module.asciidoc]: Can we also call out setting the threads as a best practice for multiple event hubs ?

File: docs/static/azure-module.asciidoc [hunk 3]
```diff
@@ -186,27 +157,67 @@ global configuration is effectively ignored.
...
+    var.input.azure_event_hubs.consumer_group: "logstash"
+    var.input.azure_event_hubs.storage_connection: "DefaultEndpointsProtocol=https;AccountName=instance1..."
     var.input.azure_event_hubs.event_hubs:
-      - ["name",                      "event_hub_connection",      "storage_connection",                                      "initial_position", "decorate_events"]
-      - ["insights-operational-logs", "Endpoint=sb://example1...", "DefaultEndpointsProtocol=https;AccountName=example1....", "HEAD",             "true"]
-      - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
-      - ["insights-logs-errors",      "Endpoint=sb://example3...", "DefaultEndpointsProtocol=https;AccountName=example3....", "TAIL",             "false"]
-      - ["insights-operational-logs", "Endpoint=sb://example4...", "DefaultEndpointsProtocol=https;AccountName=example4....", "HEAD",             "true"]
+      - ["name",                                    "initial_position",  "storage_container",  "decorate_events",  "event_hub_connection"]                                   <2>
```
Comment (jakelandis) [inline docs/static/azure-module.asciidoc]: nitpick: the `<2>` is misaligned abit

File: docs/static/azure-module.asciidoc [hunk 4]
```diff
@@ -186,27 +157,67 @@ global configuration is effectively ignored.
...
+      - ["insights-operational-logs",                 "TAIL",              "activity-logs1",    "true",             "DefaultEndpointsProtocol=https;AccountName=instance1"]
+      - ["insights-operational-logs",                 "HEAD",              "activity_logs2",<3>   "true",             "DefaultEndpointsProtocol=https;AccountName=instance2"]
```
Comment (jakelandis) [inline docs/static/azure-module.asciidoc]: nitpick: the addition of `<3>` throws off the column formatting a bit when rendered.

File: docs/static/azure-module.asciidoc [hunk 5]
```diff
@@ -368,20 +387,43 @@ Total number of threads used to process events. The value you set here applies
...
-. Be sure that the `logstash.yml` file is <<configuring-azure,configured correctly>>. 
-. Run this command from the Logstash directory:
+===== First time setup
+
+Run this command from the Logstash directory:
 
 ["source","shell",subs="attributes"]
 -----
-bin/logstash
+bin/logstash --modules azure --setup
```
Comment (jakelandis) [inline docs/static/azure-module.asciidoc]: you don't need the `--modules azure` if you have the logstash.yml configured.

File: docs/static/azure-module.asciidoc [hunk 6]
```diff
@@ -186,27 +157,67 @@ global configuration is effectively ignored.
...
+It is required for users who want to scale out multiple {ls} instances to consume from Event Hubs.
+
+A Blob Storage account is a central location that enables multiple instances of
+{ls} to work together to process events. It records the
+offset (location) of processed events. On restart, {ls} resumes processing
+exactly where it left off.
+
+Configuration notes:
+
+*  A Blob Storage account is highly recommended for use with this plugin, and is
```
Comment (jakelandis) [inline docs/static/azure-module.asciidoc]: nitpick: for use with this plugin --> for use with this module

File: docs/static/azure-module.asciidoc [hunk 7]
```diff
@@ -50,128 +52,97 @@ These {kib} dashboards are available and ready for you to use. You can use them
...
+
+Elastic products are https://www.elastic.co/downloads[available to download] and
+easy to install. 
 
 [[azure-prereqs]]
-===== Azure prereqs
+===== Azure prerequisites
 
-Azure Monitor should be configured to stream logs to one or more Event Hubs. 
+Azure Monitor should be configured to stream logs to one or more Event Hubs.
```
Comment (jakelandis) [inline docs/static/azure-module.asciidoc]: Can you link this to the [[azure-resources]] section below ?

**Comment by jakelandis** [comment]
left a few comments, but updates look really good! thanks @karenzone

File: docs/static/azure-module.asciidoc [hunk 8]
```diff
@@ -186,27 +157,67 @@ global configuration is effectively ignored.
...
-      - ["insights-metrics-pt1m",     "Endpoint=sb://example2...", "DefaultEndpointsProtocol=https;AccountName=example2....", "TAIL",             "true"]
-      - ["insights-logs-errors",      "Endpoint=sb://example3...", "DefaultEndpointsProtocol=https;AccountName=example3....", "TAIL",             "false"]
-      - ["insights-operational-logs", "Endpoint=sb://example4...", "DefaultEndpointsProtocol=https;AccountName=example4....", "HEAD",             "true"]
+      - ["name",                                    "initial_position",  "storage_container",  "decorate_events",  "event_hub_connection"]                                   <2>
+      - ["insights-operational-logs",                 "TAIL",              "activity-logs1",    "true",             "DefaultEndpointsProtocol=https;AccountName=instance1"]
+      - ["insights-operational-logs",                 "HEAD",              "activity_logs2",<3>   "true",             "DefaultEndpointsProtocol=https;AccountName=instance2"] 
+      - ["insights-metrics-pt1m",                     "TAIL",              "dbmetrics",         "true",             "DefaultEndpointsProtocol=https;AccountName=instance1..."]
```
Comment (jakelandis) [inline docs/static/azure-module.asciidoc]: the connect strings here should be Endpoint=:// (no reference to DefaultEndpointsProtocol..)

Comment (jakelandis) [reply docs/static/azure-module.asciidoc]: We should be able to use the same connection strings used in the basic example.

File: docs/static/azure-module.asciidoc (hunk 4 above)
Comment (jakelandis) [inline docs/static/azure-module.asciidoc]: nitpick: i like changing one of these from TAIL to HEAD ... but can you change a different one to HEAD other then insights-operational-logs

File: docs/static/azure-module.asciidoc (hunk 5 above)
Comment (karenzone) [reply docs/static/azure-module.asciidoc]: I'm guessing this applies to both setup and subsequent runs?  I'm removing `-modules azure` from both, so please let me know if I misinterpreted.

File: docs/static/azure-module.asciidoc (hunk 7 above)
Comment (karenzone) [reply docs/static/azure-module.asciidoc]: The link is in the next line down (line 84).  Do you want it both places?

File: docs/static/azure-module.asciidoc (hunk 8 above)
Comment (karenzone) [reply docs/static/azure-module.asciidoc]: @jakelandis  Is this what you mean? 

screen shot 2018-08-03 at 2 13 10 pm

Comment (jakelandis) [reply docs/static/azure-module.asciidoc]: the connection string should be the same between the two. ... the only that difference between the two rows should be the storage_container.

File: docs/static/azure-module.asciidoc (hunk 7 above)
Comment (jakelandis) [reply docs/static/azure-module.asciidoc]: doh! sorry , i missed that.  good as-is

File: docs/static/azure-module.asciidoc (hunk 4 above)
Comment (karenzone) [reply docs/static/azure-module.asciidoc]: Same as previous.  I'll verify alignment again after publishing.

**Comment by karenzone** [comment]
@jakelandis @acchen97  This should be it as far as recommended changes. Thanks!

File: docs/static/azure-module.asciidoc [hunk 9]
```diff
@@ -50,128 +52,98 @@ These {kib} dashboards are available and ready for you to use. You can use them
...
+      - "Endpoint=sb://...EntityPath=insights-logs-blocks"
+      - "Endpoint=sb://...EntityPath=insights-logs-databasewaitstatistics"
+      - "Endpoint=sb://...EntityPath=insights-logs-errors"
+      - "Endpoint=sb://...EntityPath=insights-logs-querystoreruntimestatistics"
+      - "Endpoint=sb://...EntityPath=insights-logs-querystorewaitstatistics"
+      - "Endpoint=sb://...EntityPath=insights-logs-timeouts"
 -----
 
+<1> The `consumer_group` (optional) is highly recommended. See <<azure_best_practices>>.
+<2> The `storage_connection` (optional) sets the Azure Blog Storage connection for tracking processing state for Event Hubs when scaling out a deployment with multiple Logstash instances. See "Scaling Event Hubs consumption" for additional details.
```
Comment (acchen97) [inline docs/static/azure-module.asciidoc]: Blog -> Blob

Comment (acchen97) [reply docs/static/azure-module.asciidoc]: @karenzone ^^

File: docs/static/azure-module.asciidoc [hunk 10]
```diff
@@ -50,128 +52,97 @@ These {kib} dashboards are available and ready for you to use. You can use them
...
```
[the remark this answers is not in the document]
Comment (acchen97) [reply docs/static/azure-module.asciidoc]: @karenzone ^^

File: docs/static/azure-module.asciidoc [hunk 11]
```diff
@@ -186,27 +157,67 @@ global configuration is effectively ignored.
 -----
```
[the remark this answers is not in the document]
Comment (acchen97) [reply docs/static/azure-module.asciidoc]: @karenzone ^^

File: docs/static/azure-module.asciidoc [hunk 12]
```diff
@@ -224,10 +237,61 @@ sure that at least one of these options is different per Event Hub:
...
+[[exploring-data-azure]]
+==== Explore your data
+When the Logstash Azure module starts receiving events, you can begin using the
+packaged Kibana dashboards to explore and visualize your data. 
+
+To explore your data with Kibana:
 
+. Open a browser to http://localhost:5601[http://localhost:5601] (username:
+  "elastic"; password: "{pwd}")
+. Click *Dashboard*.
```
Comment (acchen97) [inline docs/static/azure-module.asciidoc]: @karenzone the dashboard name is "[Azure Monitor] Overview"

File: docs/static/azure-module.asciidoc (hunk 9 above)
Comment (karenzone) [reply docs/static/azure-module.asciidoc]: Dang autocorrect from GDocs.   I've changed both instances in asciidoc source.

File: docs/static/azure-module.asciidoc (hunk 12 above)
Comment (karenzone) [reply docs/static/azure-module.asciidoc]: @acchen97 Don't you have to click "Dashboard" to see the available dashboards?  Has this changed, or am I misinterpreting you comment?

Comment (acchen97) [reply docs/static/azure-module.asciidoc]: Yes indeed, but let's include the dashboard name so they know where to navigate. :)

File: docs/static/azure-module.asciidoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (karenzone) [reply docs/static/azure-module.asciidoc]: @acchen97 Good point.  What about this:
```
In this example `threads`, `consumer_group`, and `storage_connection` will be
applied to each of the configured Event Hubs. Note that `decorate_events` is
defined in both the global and per Event Hub configuration. The per Event Hub
configuration takes precedence, and the global configuration is effectively
ignored when the per Event Hub setting is present.
```
Note that I didn't include `decorate_events` in the list.  Instead, it's behavior is addressed separately.  Let me know what you think.

File: docs/static/azure-module.asciidoc [hunk 13]
```diff
@@ -224,7 +235,15 @@ sure that at least one of these options is different per Event Hub:
...
 ** storage_container (defaults to Event Hub name if not defined)
 ** consumer_group
-
+* **Set number of threads correctly.** 
+The number of threads should equal the number of Event Hubs plus one (or more).
+Each Event Hub needs at least one thread. An additional thread is needed to help
+coordinate the other threads. The number of threads should not exceed the number of Event Hubs multiplied by the
+number of partitions per Event Hub plus one. Threads are
+currently  available only as a global setting.
+** Sample: Event Hubs = 4. Partitions on each Event Hub = 3.
```
[the remark this answers is not in the document]
Comment (karenzone) [reply docs/static/azure-module.asciidoc]: Please take a look at what I did.  I left the theoretical example to illustrate the calculation, and added your more real-world recommendation.  Let me know what you think.

**Comment by karenzone** [comment]
@acchen97 I've addressed your comments. I made my best guess on some, noted my reasoning, and invited you to respond.  Let me know what you think.

File: docs/static/azure-module.asciidoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (acchen97) [reply docs/static/azure-module.asciidoc]: ++

File: docs/static/azure-module.asciidoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (acchen97) [reply docs/static/azure-module.asciidoc]: looks good

**Comment by acchen97** [comment]
@karenzone LGTM. Good work, thank you! :)

**Pull request closed without merging**